clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -480,9 +481,6 @@ private long listOffset(TopicPartition partition, OffsetResetStrategy strategy)
     }
 
     private List<ConsumerRecord<K, V>> drainRecords(PartitionRecords<K, V> partitionRecords, int maxRecords) {
-        if (partitionRecords.isDrained())
-            return Collections.emptyList();
```

Comment:
Just to double-check, this was removed because we want to update the metrics even if `isDrained` is true?